src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -689,6 +689,114 @@ protected override void ProcessRecord()
         }
 
         #endregion Overrides
+
+        /// <summary>
+        /// Adds UserName, Description, BinPath, DelayedAutoStart and StartupType to a ServiceController object.
+        /// </summary>
+        /// <param name="service"></param>
+        /// <returns>ServiceController as PSObject with UserName, Description and StartupType added</returns>
+        private PSObject AddProperties(ServiceController service) {
+            NakedWin32Handle hScManager = IntPtr.Zero;
+            NakedWin32Handle hService = IntPtr.Zero;
+            int lastError = 0;
+            PSObject serviceAsPSObj = PSObject.AsPSObject(service);
+            IntPtr descriptionStructPtr = IntPtr.Zero;
+            IntPtr delayedAutoStartStuctPtr = IntPtr.Zero;
+            IntPtr serviceConfigStructPtr = IntPtr.Zero;
+            try
+            {
+                hScManager = NativeMethods.OpenSCManagerW(
+                    lpMachineName: service.MachineName,
+                    lpDatabaseName: null,
+                    dwDesiredAccess: NativeMethods.SC_MANAGER_CONNECT
+                );
+                if (IntPtr.Zero == hScManager) {
+                    lastError = Marshal.GetLastWin32Error();
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
+                        service,
+                        service.MachineName,
+                        exception,
+                        "ComputerAccessDenied",
+                        ServiceResources.ComputerAccessDenied,
+                        ErrorCategory.PermissionDenied);
+                }
+                hService = NativeMethods.OpenServiceW(
+                    hScManager,
+                    service.ServiceName,
+                    NativeMethods.SERVICE_QUERY_CONFIG
+                );
+                if (IntPtr.Zero == hService) {
+                    lastError = Marshal.GetLastWin32Error();
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
+                        service,
+                        exception,
+                        "CouldNotGetServiceInfo",
+                        ServiceResources.CouldNotGetServiceInfo,
+                        ErrorCategory.PermissionDenied);
+                }
+
+                bool queryStatus = NativeMethods.QueryServiceConfig2(hService, NativeMethods.SERVICE_CONFIG_DESCRIPTION, out descriptionStructPtr);
+                NativeMethods.SERVICE_DESCRIPTIONW description = (NativeMethods.SERVICE_DESCRIPTIONW)Marshal.PtrToStructure(descriptionStructPtr, typeof(NativeMethods.SERVICE_DESCRIPTIONW));
+
+                queryStatus = queryStatus & NativeMethods.QueryServiceConfig2(hService, NativeMethods.SERVICE_CONFIG_DELAYED_AUTO_START_INFO, out delayedAutoStartStuctPtr);
```

Comment:
Shouldn't this be a `|`?

Reply:
I renamed the variable to `querySuccessful` to help with the readability. I'm quite sure it's `&`. I also changed it to `&&` to make a short-circuit